plugins/notifications-backend/src/database/DatabaseNotificationsStore.ts

```diff
@@ -194,6 +233,31 @@ export class DatabaseNotificationsStore implements NotificationsStore {
       );
     }
 
+    if (options.metadata) {
+      const count = Object.keys(options.metadata ?? {}).length;
+
+      const notificationMetadataQuery = (tableName: string) => {
+        const mQuery = this.db(tableName).select('originating_id');
+
+        Object.keys(options.metadata ?? {}).forEach(key => {
+          const value = options.metadata?.[key];
+          mQuery.orWhereRaw(
+            `(LOWER(name) LIKE LOWER(?) AND LOWER(value) LIKE LOWER(?))`,
```

Comment:
just noting that this will likely bust the index entirely, and use a sequential scan instead

Reply:
What can be an alternative to it?
I reused the same query which is used for querying the description 😅

Reply:
storing pre-lowercased values and then comparing against a lowercased value, for example - but these are likely to very small datasets right? if so, it's not a big deal

Reply:
Oh, yeah, that's really a good idea, as an extra column.
For our company I envision with some time we might have millions of records, so better to think about the optimizations. I wasn't aware that LOWER does influence the index